src/components/DragAndDrop/Consumer/index.js

```diff
@@ -0,0 +1,35 @@
+import React, {useContext, useEffect, useRef} from 'react';
+import {Portal} from '@gorhom/portal';
+import dragAndDropConsumerPropTypes from './dragAndDropConsumerPropTypes';
+import DragAndDropUtils from '../Utils';
+
+function DragAndDropConsumer({children, dropZoneID, onDrop}) {
+    const DragAndDropContext = DragAndDropUtils.getDragAndDropContext(dropZoneID);
+    const {isDraggingOver} = useContext(DragAndDropContext);
+
+    const onDropRef = useRef(onDrop);
+    onDropRef.current = onDrop;
```

Comment:
Yeah, this is a good call-out. I think this pattern makes sense when you want to avoid adding and removing listeners over and over or (especially) if what you're doing is going to affect the DOM.

But in this case deregistering and re-registering should be prettymuch negligible and not have side-effects, so we might be able to remove this.

Let me test...

Reply:
I changed it for the sake of simplicity, though FWIW I don't think the ref approach I had here was "wrong" at all, just a premature performance optimization.